OpenTelemetry Project Security Self-Assessment

PrathamGupta opened the pull request:
Created and added first draft of the [OpenTelemetry](https://opentelemetry.io) Project Security Self-Assessment.
Please feel free to share any thoughts on the self-assessment.

eddie-knight commented:
Hi there! I'm just getting started looking at your pull request, and I noticed the DCO check is failing.

You can look at the checks section of the PR (I believe it should always be below the last comment) and look for a red X highlighting the failed check. In this case, you can click `Details` for more information about how to get that check passing.
<img width="315" alt="Screenshot 2023-12-08 at 8 35 18 AM" src="https://github.com/cncf/tag-security/assets/21176439/501ae793-0173-4e38-8ea2-57dca65ab1ea">

eddie-knight commented on assessments/projects/open-telemetry/self-assessment-template.md:
```diff
@@ -0,0 +1,173 @@
+# Self-assessment
```
```suggestion
# Open Telemetry Self-assessment
```

ragashreeshekar reviewed (changes requested):
Thanks for the PR @PrathamGupta and team, appreciate the efforts.
I have completed first pass of review. Please feel free to reach out here or on slack for any questions and clarifications.

Along with addressing the comments, kindly update the PR branch with the latest content in the repo as this branch is out-of-date with the base branch.

PrathamGupta commented on assessments/projects/open-telemetry/self-assessment-template.md:
We have updated our PR with removing all the template content and tailored it for these needs. We have also corrected all the checks and will keep the PR updated for these checks in the future.

PrathamGupta commented:
> Thanks for the PR @PrathamGupta and team, appreciate the efforts. I have completed first pass of review. Please feel free to reach out here or on slack for any questions and clarifications.
> 
> Along with addressing the comments, kindly update the PR branch with the latest content in the repo as this branch is out-of-date with the base branch.

Thank you @ragashreeshekar for reviewing our work! We have updated the PR branch with the latest content in the repo and will be updated regularly till the PR has been reviewed completely.

PrathamGupta commented on assessments/projects/open-telemetry/self-assessment.md:
```diff
@@ -0,0 +1,340 @@
+
+* Use API tokens and TLS network security encryption to encrypt sensitive information to prevent third parties from querying OpenTelementry telemetry data.
+* Prevent external access to internal resources. When running Open Telemetry, there should be no privileged access.
+* The agents must authenticate themselves to the backend systems where the data is sent.
```
We have updated the same for Non Goals and we will also try to update the goals and add more details.

PrathamGupta commented on assessments/projects/open-telemetry/self-assessment.md:
```diff
@@ -0,0 +1,340 @@
+* High performance: Unexpected interference effects do not exist in the host application.
+* Simplicity: It is only necessary to learn a single set of APIs and conventions to capture traces and metrics, simplifying the application instrumentation process.
+
+**Security**
```
We have updated the goals of the document and increased the scope in this section.

Thanks for the feedback

PrathamGupta commented on assessments/projects/open-telemetry/self-assessment.md:
Noted and changed.

Thanks for the feedback!

PrathamGupta commented on assessments/projects/open-telemetry/self-assessment.md:
The things which are not applicable are removed. 

Thanks for feedback!

PrathamGupta commented on assessments/projects/open-telemetry/self-assessment.md:
We have added this as a recommendation for the project!

PrathamGupta commented:
@ragashreeshekar We have updated the PR with updates from your suggestions! Please let us know your thoughts about the assessment now! 

Thanks!

PrathamGupta commented:
@torinvdb We are currently in the process of making the changes you have suggested and going through your feedback

PrathamGupta commented on assessments/projects/open-telemetry/self-assessment.md:
```diff
@@ -0,0 +1,340 @@
+# OpenTelemetry Security Self Assessment
+
```
We have added this disclaimer again! We removed it on our assumption that this was not necessary, but we know understand the use of this disclaimer!

Thanks for the feedback

PrathamGupta commented on assessments/projects/open-telemetry/self-assessment.md:
We have added the reference to our assessment as well

PrathamGupta commented on assessments/projects/open-telemetry/self-assessment.md:
```diff
@@ -0,0 +1,340 @@
+
+### Background
+
+OpenTelemetry is an Observability framework and toolkit designed to create and manage telemetry data such as traces, metrics, and logs. Opentelemetry provides a standard protocol for dealing with telemetry data, language SDKs, a collector that receives, processes, and exports telemetry data, as well as other documentation and tools for telemetry.
```
We have changed this in our assessment

PrathamGupta commented on assessments/projects/open-telemetry/self-assessment.md:
```diff
@@ -0,0 +1,340 @@
+
+Furthermore, a third party application that produces data in the OTLP format may be used to send telemetry to the collector.
+
+As these components are the primary source of incoming data, they also serve as the primary attack surface. A hypothetical attacker may be able to deliver a malicious payload by simply performing an action that is logged. They can use this to either attack the SDK and try and gain control of the system directly, or to attack components further down the pipeline, such as the Collector.
```
We have reviewed and changed this as well.

PrathamGupta commented on assessments/projects/open-telemetry/self-assessment.md:
```diff
@@ -0,0 +1,340 @@
+
+**OpenTelemetry Collector**
+
+The OpenTelemetry Collector is a vendor-agnostic implementation of how to receive, process and export telemetry data. It consists of multiple components: Receivers, Processors, Exporters, and Extensions. It serves as middleware between the raw telemetry feeds and telemetry frontends. It receives data via the Receiver using the OpenTelemetry Protocol. The collector then processes the data, handlin ...
```
We shall change this in other sections as well

PrathamGupta commented on assessments/projects/open-telemetry/self-assessment.md:
```diff
@@ -0,0 +1,340 @@
+
+## Self-Assessment Use
+
+Our team comprises graduate students from different professional backgrounds, and each member is rich in insight. We have come together with a shared commitment to create this self-assessment to analyze the overall security of OpenTelemetry. This document mainly helps users get a preliminary understanding of OpenTelemetry’s security architecture and security development practices. In addition, it ...
+
+All in all, this document helps explain this CNCF open source project, pave the way for the project to seek graduation, provide clear guidance during the incubation process, and bring a more transparent and clear security audit to the overall project.
```
Thank you for the insights regarding this and we have changed this in our assessment as well

PrathamGupta commented on assessments/projects/open-telemetry/self-assessment.md:
We have added one example of use case in Kubernetes from the specifications mentioned on the websites. Let us know if we should add for more of the integrations

PrathamGupta commented on assessments/projects/open-telemetry/self-assessment.md:
```diff
@@ -0,0 +1,340 @@
+
+The data is analyzed, and insights are presented through dashboards or alerts. Access to these insights is typically controlled through user authentication and authorization, ensuring that only authorized users can view or manipulate sensitive telemetry data.
+
+**Alerting and Incident Response**
+
+In case of anomalies or incidents detected through telemetry data, alerts are generated. The alerting mechanism is secured to prevent false alerts and ensure that alerts reach the correct recipients. Incident response protocols are in place to handle any security incidents reported through these alerts.
```
We have ammended the same and answered some of the questions to our best ability.

torinvdb commented on assessments/projects/open-telemetry/self-assessment.md:
```diff
@@ -0,0 +1,376 @@
+# OpenTelemetry Security Self Assessment
+
+The Self-assessment is the initial document for projects to begin thinking about the security of the project, determining gaps in their security, and preparing any security documentation for their users. This document is ideal for projects currently in the CNCF sandbox as well as projects that are looking to receive a joint assessment and currently in CNCF **Incubation**.
```
```suggestion
# OpenTelemetry Security Self Assessment

This assessment was created by community members as part of the [Security Pals](https://github.com/cncf/tag-security/issues/1102) process and is currently pending changes from the maintainer team.

The Self-assessment is the initial document for projects to begin thinking about the security of the project, determining gaps in their security, and preparing any security documentation for their users. This document is ideal for projects currently in the CNCF sandbox as well as projects that are looking to receive a joint assessment and currently in CNCF **Incubation**.
```
@ragashreeshekar

JustinCappos commented on assessments/projects/open-telemetry/self-assessment.md:
```suggestion
```

JustinCappos reviewed (approved):
LGTM.  Ready to merge!
